testutils/src/main/scala/za/co/absa/enceladus/testutils/infoFileComparison/AtumModelUtils.scala

```diff
@@ -0,0 +1,196 @@
+    * @param metadata ControlMeasureMetadata instance
+    */
+  implicit class ControlMeasureMetadataOps (metadata: ControlMeasureMetadata){
+    /**
+      * Compare this ControlMeasureMetadata with the one passed in
+      * @param otherMetadata Second control measure metadata
+      * @param curPath Path to the ControlMeasureMetadata through the model
+      * @return Returns a list of model differences
+      */
+    def compareWith(otherMetadata: ControlMeasureMetadata, curPath: String): List[ModelDifference[_]] = {
+      val diffs = simpleCompare(metadata.sourceApplication,
+        otherMetadata.sourceApplication, s"$curPath.sourceApplication") ::
+      simpleCompare(metadata.country, otherMetadata.country, s"$curPath.country") ::
+      simpleCompare(metadata.historyType, otherMetadata.historyType, s"$curPath.historyType") ::
+      simpleCompare(metadata.dataFilename, otherMetadata.dataFilename, s"$curPath.dataFilename") ::
+      simpleCompare(metadata.sourceType, otherMetadata.sourceType, s"$curPath.sourceType") ::
+      simpleCompare(metadata.version, otherMetadata.version, s"$curPath.version") ::
+      simpleCompare(metadata.informationDate, otherMetadata.informationDate, s"$curPath.informationDate") ::
+        Nil
+
+      val additionalInfoDiff = additionalInfoComparison(metadata.additionalInfo,
+        otherMetadata.additionalInfo,
+        s"$curPath.additionalInfo")
+
+      (diffs flatten) ::: additionalInfoDiff
+    }
+
+    /**
+      * Compare additional information from ControlMeasureMetadata
+      * @param was     Value it had in reference
+      * @param is Value it has now
+      * @param curPath Path to the AdditionalInfo through the model
+      * @return Returns a list of model differences
+      */
+    private def additionalInfoComparison(was: Map[String, String],
+                                         is: Map[String,String],
+                                         curPath: String): List[ModelDifference[_]] = {
+      was.flatMap {
+        case (wasKey, wasValue) if wasKey == "conform_enceladus_version" =>
+          logVersionAndContinue("Conformance", wasValue, is.get(wasKey))
+        case (wasKey, wasValue) if wasKey == "std_enceladus_version" =>
+          logVersionAndContinue("Standartization", wasValue, is.get(wasKey))
+        case (wasKey, _) if wasKey == "std_application_id" || wasKey == "conform_application_id" => None
+        case (wasKey, wasValue) =>
+          is.get(wasKey) match {
+            case Some(isValue) if wasValue != isValue => Some(ModelDifference(s"$curPath.$wasKey", wasValue, isValue))
+            case None                                 => Some(ModelDifference(s"$curPath.$wasKey", wasValue, "Null"))
+            case _                                    => None
+          }
+      }.toList
+    }
+
+    private def logVersionAndContinue(name: String, refVersion: String, newVersion: Option[String]): Option[Nothing] = {
```

Comment:
I'd suggest this function return a `Unit` and insert explicit `None` to pattern matching lines.
The reason being 2 surprises that gave me a hard time reading the code:
1. I was wondering why several cases invoke `logVersionAndContinue()` while others return an Optional.
2. `Optional[Nothing]` is a surprise by itself. Looks like a boolean, but for some reason, it is not :)